src/app/pre-sale/components/popup/index.tsx

```diff
@@ -1,24 +1,45 @@
 import { Modal, ModalBody, ModalContent, ModalOverlay } from '@chakra-ui/react'
 
-import { PreSaleWidget } from '../pre-sale-widget'
+import { PreSaleTokenCard } from '@/app/pre-sale/components/pre-sale-token-card'
+import { getDefaultIndex } from '@/lib/utils/tokens'
+
+import { DepositProvider } from '../../providers/deposit-provider'
 import { PreSaleToken } from '../../types'
+import { PreSaleWidget } from '../pre-sale-widget'
 
 type PreSalePopupProps = {
   isOpen: boolean
   onClose: () => void
+  // TODO: use token object to pre sale token interface to pass to provider?
   token: PreSaleToken | null
 }
 
 export const PreSalePopup = (props: PreSalePopupProps) => {
-  const { isOpen, onClose } = props
+  const { isOpen, onClose, token } = props
   return (
-    <Modal onClose={onClose} isOpen={isOpen} isCentered scrollBehavior='inside'>
+    <Modal
+      onClose={onClose}
+      isOpen={isOpen}
+      isCentered
+      size='full'
+      scrollBehavior='inside'
+    >
       <ModalOverlay className='bg-ic-black bg-opacity-60 backdrop-blur' />
       <ModalContent className='bg-transparent shadow-none'>
         <ModalBody className='bg-transparent'>
-          <div className='flex flex-row'>
-            <PreSaleWidget />
-          </div>
+          <DepositProvider preSaleToken={getDefaultIndex()}>
+            <div className='flex h-screen'>
+              <div className='align-center max-h-4xl mx-auto my-auto flex max-w-4xl flex-row items-start gap-3'>
+                {token && (
+                  <PreSaleTokenCard
+                    token={token}
+                    onClick={() => console.log('join')}
+                  />
+                )}
+                <PreSaleWidget token={token!} />
+              </div>
+            </div>
+          </DepositProvider>
```

Comment:
Added here because I think it would be nice to have an additional PreSaleProvider (or named similar) to handle kind of all info related things (wrapping the whole page). And then then the DepositProvider would be just about depositing/withdrawing. Should keep the providers readable with less lines and less responsibilites.